diff --git a/src/agb/image.py b/src/agb/image.py
index 59a8ece..7a5ecd8 100644
--- a/src/agb/image.py
+++ b/src/agb/image.py
@@ -202,7 +202,7 @@ def from_file(file_path: str) -> tuple[Image, Palette]:
         reader = png.Reader(f)
         width, height, data, attributes = reader.read() # type: ignore
         attributes: dict[str, int | str] = attributes
-        assert attributes['bitdepth'] in (4, 8), 'Invalid bitdepth'
+        assert attributes['bitdepth'] in (2, 4, 8), 'Invalid bitdepth'
         image = Image(None, width, height, depth=attributes['bitdepth'])
         image.data = np.array([*data]).T
         colors = attributes['palette']
