diff --git a/src/agb/model/type.py b/src/agb/model/type.py
index 8561806..1231ea8 100644
--- a/src/agb/model/type.py
+++ b/src/agb/model/type.py
@@ -196,7 +196,7 @@ class Type(ABC):
 
 def associate_with_constant(
     value: ScalarModelValue, proj: Project, constant: str | None
-) -> int | str:
+) -> ScalarModelValue:
     """Tries to associate a value form a constant table of the pymap project.
 
     Parameters:
@@ -222,7 +222,6 @@ def associate_with_constant(
             warn(f'No match for value {value} found in constant table {constant}')
         else:
             warn(f'Constant table {constant} not found in project.')
-    assert isinstance(value, int)
     return value
 
 
